core/lib/config/src/configs/object_store.rs

```diff
@@ -1,19 +1,34 @@
 use serde::Deserialize;
 
-#[derive(Debug, Deserialize, Eq, PartialEq, Clone, Copy)]
-pub enum ObjectStoreMode {
-    GCS,
-    GCSWithCredentialFile,
-    FileBacked,
-    GCSAnonymousReadOnly,
-}
-
 /// Configuration for the object store
-#[derive(Debug, Deserialize, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct ObjectStoreConfig {
-    pub bucket_base_url: String,
+    #[serde(flatten)]
     pub mode: ObjectStoreMode,
-    pub file_backed_base_path: String,
-    pub gcs_credential_file_path: String,
+    #[serde(default = "ObjectStoreConfig::default_max_retries")]
     pub max_retries: u16,
 }
+
+impl ObjectStoreConfig {
+    const fn default_max_retries() -> u16 {
+        5
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[serde(tag = "mode")]
+pub enum ObjectStoreMode {
```

Comment:
Sort of unrelated change, but I've found `ObjectStoreConfig` as-is overly convoluted and not maintainable. The new format makes invalid states non-representable. I *think* the change is backward-compatible (other than the associated Protobuf declarations, but AFAIU, they aren't used yet).